db/python/layers/sample.py

```diff
@@ -243,8 +243,8 @@ async def upsert_sample(
                         participant_id=s.participant_id,
                         type_=s.type,
                         active=s.active,
-                        sample_parent_id=sample_parent_id,
-                        sample_root_id=sample_root_id,
+                        sample_parent_id=r.parent.id if r.parent else sample_parent_id,
```

Comment:
Not strictly relevant to this PR, but a bug from #840

Reply:
Fair enough!